bears/python/RadonBear.py

```diff
@@ -18,39 +19,57 @@ class RadonBear(LocalBear):
     CAN_DETECT = {'Complexity'}
 
     def run(self, filename, file,
+            cyclomatic_complexity: int = None,
             radon_ranks_info: typed_list(str) = (),
             radon_ranks_normal: typed_list(str) = ('C', 'D'),
             radon_ranks_major: typed_list(str) = ('E', 'F'),
             ):
         """
         Uses radon to compute complexity of a given file.
 
-        :param radon_ranks_info:   The ranks (given by radon) to
-                                   treat as severity INFO.
-        :param radon_ranks_normal: The ranks (given by radon) to
-                                   treat as severity NORMAL.
-        :param radon_ranks_major:  The ranks (given by radon) to
-                                   treat as severity MAJOR.
+        :param cyclomatic_complexity: Maximum cyclomatic complexity
+                                      that is considered to be normal.
+        :param radon_ranks_info:      The ranks (given by radon) to
+                                      treat as severity INFO.
+        :param radon_ranks_normal:    The ranks (given by radon) to
+                                      treat as severity NORMAL.
+        :param radon_ranks_major:     The ranks (given by radon) to
+                                      treat as severity MAJOR.
         """
         severity_map = {
             RESULT_SEVERITY.INFO: radon_ranks_info,
             RESULT_SEVERITY.NORMAL: radon_ranks_normal,
             RESULT_SEVERITY.MAJOR: radon_ranks_major
         }
+
+        if cyclomatic_complexity is None:
+            logging.warning('The settings `radon_ranks_info`, '
+                            '`radon_ranks_normal` and `radon_ranks_major`'
+                            ' are deprecated. Please use '
+                            '`cyclomatic_complexity` instead.')
+
         for visitor in radon.complexity.cc_visit(''.join(file)):
             rank = radon.complexity.cc_rank(visitor.complexity)
             severity = None
             for result_severity, rank_list in severity_map.items():
                 if rank in rank_list:
                     severity = result_severity
-            if severity is None:
+
+            if (cyclomatic_complexity and
+                    visitor.complexity <= cyclomatic_complexity):
                 continue
+            elif (cyclomatic_complexity is None and
+                    severity is None):
+                continue
+
+            if cyclomatic_complexity is not None:
+                severity = RESULT_SEVERITY.MAJOR
```

Comment:
oh, I think this is going to be a regression.

previously the user could have log entries about ranks between certain levels, and choose which severity level is an error.

See [`--min-severity`](https://github.com/coala/coala/blob/master/coalib/parsing/DefaultArgParser.py#L192)

And I believe severity can also be a per section setting, but we had an issue about this (might have been a documentation issue).

The quick fix here is that when `cyclomatic_complexity` is used (and should be be `cyclomatic_complexity_max`?), any result above that should be major, and any other result should be info level.  Then the coala engine should eliminate the info level results by default.

Reply:
At various places in coala like [ClangComplexityBear](https://github.com/coala/coala-bears/blob/master/bears/c_languages/ClangComplexityBear.py), [JSHintBear](https://github.com/coala/coala-bears/blob/master/bears/js/JSHintBear.py), etc the setting max-complexity or any other such name was deprecated and replaced with cyclomatic_complexity. Since, the main aim was to maintain the consistency, I chose to adopt the setting name as cyclomatic-complexity.